Tests/BasicsTests/NetrcTests.swift

```diff
@@ -431,5 +431,80 @@ class NetrcTests: XCTestCase {
         XCTAssertEqual(netrc.machines[1].login, "fred")
         XCTAssertEqual(netrc.machines[1].password, "sunshine4ever")
     }
-}
 
+    func testComments() throws {
+        let content = """
+            # A comment at the beginning of the line
+            machine example.com # Another comment
+            login anonymous
+            password qw#erty
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw#erty")
+    }
+
+    func testHashSymbolInPassword() throws {
+        let content = """
+            machine example.com
+            login anonymous
+            password qw#erty
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw#erty")
+    }
+
+    func testQuotedPasswordWithSpace() throws {
+        let content = """
+            machine example.com
+            login anonymous
+            password "qw erty"
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw erty")
+    }
+
+    func testQuotedPasswordWithSpaceAndHash() throws {
+        let content = """
+            machine example.com
+            login anonymous
+            password "qw #erty"
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw #erty")
+    }
+
+    func testQuotedPasswordWithSpaceAndHashAndCommentAtEndOfLine() throws {
```

Comment:
suggestion (blocking): the more I read about .netrc, the more I see test gaps :S. can we add a test case sphere the machine, login and password are all on a single line,  them add the same with comments.

Also (non-blocking), is there value is describing the behaviour of the parser should a `#` character appear in the value of `login` and/or `machine`?   This will probably never happen, but maybe we should document the parsers expected behaviour.

Reply:
I've reworked the tests to cover these cases, and made a note to convert them to parameterized tests when we switch to swift-testing.

Reply:
This PR may conflict with #8093.  So depending on which is merge first, the other will need to be updated to be Swift Testing compatible :)

Reply:
That should be fixed now: https://ci.swift.org/job/swift-package-manager-with-xcode-self-hosted-PR-osx/4511/execution/node/49/log/